Sources/SuperwallKit/Superwall.swift

```diff
@@ -0,0 +1,354 @@
+  ///
+  /// If you haven't called ``SuperwallKit/Superwall/logIn(userId:)`` or ``SuperwallKit/Superwall/createAccount(userId:)``,
+  /// this value will return an anonymous user id which is cached to disk
+  public static var userId: String {
+    return shared.dependencyContainer.identityManager.userId
+  }
+
+  /// Indicates whether the user is logged in to Superwall.
+  ///
+  /// If you have previously called ``SuperwallKit/Superwall/logIn(userId:)`` or
+  /// ``SuperwallKit/Superwall/createAccount(userId:)``, this will return true.
+  ///
+  /// - Returns: A boolean indicating whether the user is logged in or not.
+  public static var isLoggedIn: Bool {
+    return shared.dependencyContainer.identityManager.isLoggedIn
+  }
+
+  /// A published property that indicates whether the device has any active subscriptions.
+  ///
+  /// Its value is stored on disk and synced with the active purchases on device.
+  ///
+  /// If you're using Combine or SwiftUI, you can subscribe or bind to this to get
+  /// notified whenever the user's subscription status changes.
+  ///
+  /// If you are returning a ``SubscriptionController`` in the
+  /// ``SuperwallDelegate``, you should rely on your own subscription status instead.
+  @Published
+  public var hasActiveSubscription = false {
+    didSet {
+      dependencyContainer.storage.save(hasActiveSubscription, forType: SubscriptionStatus.self)
+    }
+  }
+
+  /// A published property that is `true` when Superwall has finished configuring via
+  /// ``configure(apiKey:delegate:options:)-65jyx``.
+  ///
+  /// If you're using Combine or SwiftUI, you can subscribe or bind to this to get
+  /// notified whenever the user's subscription status changes.
+  @Published
+  public var isConfigured = false
+
+  /// The configured shared instance of ``Superwall``.
+  ///
+  /// - Warning: This method will crash with `fatalError` if ``Superwall`` has
+  /// not been initialized through ``configure(apiKey:delegate:options:)-65jyx``.
+  /// If there's a chance that may have not happened yet, you can use
+  /// ``isConfigured`` to check if it's safe to call.
+  /// ### Related symbols
+  /// - ``isConfigured``
+  @objc(sharedSuperwall)
+  public static var shared: Superwall {
+    guard let superwall = superwall else {
+      #if DEBUG
+      // Code only executes when tests are running in a debug environment.
+      // This avoids lots of irrelevent error messages printed to console about Superwall not
+      // being configured, which slows down the tests.
+      if ProcessInfo.processInfo.environment["XCTestConfigurationFilePath"] != nil {
+        return Superwall()
+      }
+      #endif
+      Logger.debug(
+        logLevel: .error,
+        scope: .superwallCore,
+        message: "Superwall has not been configured. Please call Superwall.configure()"
+      )
+      return Superwall()
+    }
+    return superwall
+  }
+
+  // MARK: - Internal Properties
+  private static var superwall: Superwall?
```

Comment:
This idea was also taken from RevenueCat's framework. As an end user, I feel like it might be really annoying if you have to always check whether `Superwall.shared` isn't `nil` when binding to `hasActiveSubscription` or `isConfigured`. I saw Mixpanel's shared instance isn't `nil` either. So I'm kinda opposed to changing this. We can add a comment to `superwall` so that internally it's clear you shouldn't use it in that file.
Btw it doesn't actually crash if you use Superwall.shared without configuring, it just prints out an error log. RevenueCat do actually make it crash

Reply:
Haha! I was looking back at this again and so confused as to why I thought it would crash – glad you saw that.

I agree on the optionality being annoying. Your comments seem reasonable.